### DISCUSSION TITLE
How to search for a user and display a user's profile on Nostr Console

### DISCUSSION THREAD

**Discussion by vishalxl** [vishalxl/nostr_console#25] [General]
1. Nostr console allows you to search a pubkey by just first  few letters. Example, searching for b23, shows result as shown in the pic: 

<img width="1280" alt="2022-11-13" src="https://user-images.githubusercontent.com/64505169/201496409-868d1b56-5406-4edd-8fed-114545fe4d7b.png">

**Comment by NorrinRadd** [comment]
![pic](https://user-images.githubusercontent.com/971294/201498265-2ad087bd-0dd5-49a3-9edb-15cc87603fc0.png)

I don't have a way to not which pub key is the one I want (the one that is actually you).  All of the other UI only shows your username and not pub key.  So I am not informed which pub key corresponds to the messages that I've been reading.

**Comment by vishalxl** [comment]
The new [version 0.1.3-beta](https://github.com/vishalxl/nostr_console/releases/tag/v0.1.3-beta) is now printing the names with the pubkeys in this scenario.

<img width="1280" alt="2022-11-14 (1)" src="https://user-images.githubusercontent.com/64505169/201542725-983982d7-f07b-49bb-aede-be7dd9bc528f.png">

**Reply by NorrinRadd** [reply]
So in this scenario, which one of these pairs is the JB55 that most people know? The issue is that when you view the public feed, or a channel, there is no way to see the pub key of the poster. You only see a username that may not be unique. A potential fix could be a menu option available where you could see the raw data about an event.  You enter the event ID there, and it can display the pub key of the poster, in addition to lots of other information that is usually not in the UI such as 'kind' number and whatever else. I hope this helps.

**Comment by vishalxl** [comment]
jb55 is an extreme example. he has made many accounts with same name. right now such cases are rare. If you know a name, you'd typically find their pubkey directly. Even if 2-3 are shown, one can look at each profile to see who they want. But long term it will need a solution. will  just keep that in mind for now.

**Reply by NorrinRadd** [reply]
ok i see. Yes, the profiles don't really show proof of who is who right now. I can copy the details of someone else's profile right now and they would be identical. Only signed messages show proof of who is who, hence the events would have to be visible, or, as mentioned before, one would need to be able to get the raw details of an event on the primary feed or channel.

**Comment by vishalxl** [comment]
How user id's are shown now, NIP 05 status is mentioned if they're verified. 

<img width="1280" alt="2022-11-16 (1)" src="https://user-images.githubusercontent.com/64505169/202095800-0de2d848-d6e1-4b89-b57f-298118349d9b.png">

**Comment by vishalxl** [comment]
Added about and nip05 info too when printing user account information in this case: 

<img width="1280" alt="2022-11-22 (8)" src="https://user-images.githubusercontent.com/64505169/203244782-75362f07-65ef-4df1-91b6-6f36a2fa687f.png">
